### ISSUE TITLE
Denormalization bad behaviour

### ISSUE THREAD

**Issue by alanzanattadev** [paularmstrong/normalizr#248]
Hi

# Problem

I'm trying to denormalize some data, basically, the same way as the example, but I don't get a similar result at all.

**Input**

Here's how I'm using normalizr:

```js
const input = { Patient: patients.keySeq().toJS().map(v => parseInt(v)) };
  const schema = { Patient: [normalizrModel.entities.Patient] };
  const entities = selectCache(state).toJS();
  const denormalized = denormalize(input, schema, entities);
  console.log("-------------------");
  console.log(JSON.stringify(input));
  console.log(schema);
  console.log(JSON.stringify(entities));
  console.log(JSON.stringify(denormalized));
```

```js

const entities = {"CareSeeker":{"1":{"id":1}},"Auction":{"undefined":{"status":1,"new_responses":6}},"Patient":{"1":{"id":1,"user_id":"AAA-AAA-AAA","auction_status":1,"careseeker":1,"profile":{"mobile_care_requirements":{"base_care":null},"static_care_requirements":null,"discharge_date":1487084810,"careseeker_data":{"social_worker_name":"Social worker name","division":null,"station":"Station"}}},"2":{"id":2,"user_id":"BBB-BBB-BBB","auction_status":1,"careseeker":1,"profile":{"mobile_care_requirements":{"base_care":null},"static_care_requirements":null,"discharge_date":1487084810,"careseeker_data":{"social_worker_name":"Social worker name","division":null,"station":"Station"}}}}}

const input = {"Patient":[1,2]}

```

Schema is generated so this is a console.log of it

![screenshot from 2017-03-06 16-40-51](https://cloud.githubusercontent.com/assets/9500204/23616980/c737b5de-028b-11e7-9d7a-821535b95f7b.png)

**Output**

Here's what I expect to see when I run the above:

```js
{
  Patient: [{
    id: 1,
    others attributs
  }, {
     id: 2,
     others attributs
  }]
}
```

Here's what I *actually* see when I run the above:

```js
{
    Patient: [1, 2]
}
```

It seems like it doesn't try to resolve id as it should, and the structure is exactly the same as the denormalize example. Maybe I've misunderstood something, but I don't think so

Thank you :)

**Comment by paularmstrong** [comment]
@alanzanattadev is this the same as #240? I'm looking for help fixing that, as I'm a bit swamped and not really using Immutable on anything to know what needs to be done.

**Comment by alanzanattadev** [comment]
Weird because I call toJS to be sure not entering any issue with Immutable for now but yes it seems to be the same bad result. How can I help ? Do you have some ideas of possible source ?

**Comment by azangru** [comment]
@paularmstrong I have come across this problem while normalizing and denormalizing data using arrays of union schemas. I have captured this unexpected behavior in a couple of tests that I have submitted via [this PR](https://github.com/paularmstrong/normalizr/pull/252). It seems that while javascript arrays work reliably, the schema.Array constructor produces incorrect results during denormalization.

I am afraid I cannot offer the solution to the problem though :-(
